src/content/transformers/contentPromptsQuestionsTransformer.ts

```diff
@@ -0,0 +1,26 @@
+import { Logger } from '@nestjs/common';
+import {
+  Transformer,
+  AIContentTemplateRaw,
+  ContentPromptsQuestions
+} from '../content.types';
+import { camelCase } from 'lodash';
+
+const logger = new Logger('ContentPromptsQuestionsTransformer');
+
+export const contentPromptsQuestionsTransformer: Transformer<
+  AIContentTemplateRaw,
+  ContentPromptsQuestions
+> = (templates: AIContentTemplateRaw[]): ContentPromptsQuestions[] => {
+  const result = templates.reduce<ContentPromptsQuestions>((acc, template) => {
+    if (template.data.name && template.data.requiresAdditionalQuestions) {
+      const name = camelCase(template.data.name);
+      acc[name] = template.data.requiresAdditionalQuestions;
+    } else {
+      logger.warn('template.data.name and/or template.data.requiresAdditionalQuestions not found', template);
+    }
+    return acc;
+  }, {});
+
+  return [result];
+};
```

Comment:
The logic here, and in `candidateContentPromptsTransformer` are essentially exactly the same, are they not?

Can we please reconcile and reuse the logic of the one for the other in some way? 🤔

Reply:
I looked into this, and it just seems to create an excessive abstraction that results in more code. Let me know if I'm missing something

```import { Logger } from '@nestjs/common';
import { Transformer, AIContentTemplateRaw } from '../content.types';
import camelCase from 'just-camel-case';

function createTransformer<T>(
  loggerContext: string,
  extractor: (data: AIContentTemplateRaw['data']) => [string, T] | null,
): Transformer<AIContentTemplateRaw, Record<string, T>> {
  const logger = new Logger(loggerContext);

  return (templates: AIContentTemplateRaw[]): Record<string, T>[] => {
    const result = templates.reduce<Record<string, T>>((acc, { data }) => {
      const extracted = extractor(data);
      if (extracted) {
        const [key, value] = extracted;
        acc[camelCase(key)] = value;
      } else {
        logger.warn('Required fields missing', data);
      }
      return acc;
    }, {});
    return [result];
  };
}

export const contentPromptsQuestionsTransformer = createTransformer(
  'ContentPromptsQuestionsTransformer',
  (data) =>
    data.name && data.requiresAdditionalQuestions
      ? [data.name, data.requiresAdditionalQuestions]
      : null,
);

export const candidateContentPromptsTransformer = createTransformer(
  'CandidateContentPromptsTransformer',
  (data) =>
    data.name && data.content ? [data.name, data.content] : null,
);```

Reply:
Ah.. I see what you're saying now.. yeah.. looks like an unhelpful exercise